docs/pages/management/operations/enroll-agent-into-automatic-updates.mdx

```diff
@@ -0,0 +1,225 @@
+
+```code
+$ touch /etc/teleport-upgrade.d/schedule
+$ chown <your-teleport-user> /etc/teleport-upgrade.d/schedule
+```
+
+Configure the updater to connect to your custom version server and subscribe
+to the right release channel:
+
+```code
+$ echo <version-server-url/path>/<release-channel> > /etc/teleport-upgrade.d/endpoint
+```
+
+Install the `teleport-ent-updater` package (note: your teleport agent will be restarted during install):
+
+```code
+$ apt install teleport-ent-updater
+# or
+$ yum install teleport-ent-updater
+```
+
+Finally, verify that the updater can see your version endpoint:
+
+```code
+$ teleport-upgrade dry-run
+```
+
+You should see one of the following messages, depending on the target version
+you are currently serving:
+
+```text
+no upgrades available (1.2.3 == 1.2.3)
+an upgrade is available (1.2.3 -> 2.3.4)
+```
+
+<Admonition type="note">
+`teleport-upgrade` may complain about not having a valid upgrade schedule.
+This is expected immediately after install as the maintenance schedule might
+not be exported yet.
+</Admonition>
+
+</TabItem>
+
+<TabItem label="teleport-kube-agent chart" options="Self-hosted">
+
+Add the following chart values to your existing agent `values.yaml`:
+
+```yaml
+updater:
+  enabled: true
+  versionServer: https://<version-server-domain-and-path>
+  releaseChannel: <release-channel>
+```
+
+Update the Helm chart release with the new values by running `helm upgrade`.
+
+You can validate the updater is running properly by checking if its pod is ready:
+
+```code
+$ kubectl get pods
+NAME                               READY   STATUS    RESTARTS   AGE
+my-agent-0                         1/1     Running   0          14m
+my-agent-1                         1/1     Running   0          14m
+my-agent-2                         1/1     Running   0          14m
+my-agent-updater-d9f97f5dd-v57g9   1/1     Running   0          16m
+```
+
+And by consulting its logs:
+
+```code
+$ kubectl logs <your-agent-release>-updater
+2023-04-28T13:13:30Z	INFO	StatefulSet is already up-to-date, not updating.	{"controller": "statefulset", "controllerGroup": "apps", "controllerKind": "StatefulSet", "StatefulSet": {"name":"my-agent","namespace":"agent"}, "namespace": "agent", "name": "my-agent", "reconcileID": "10419f20-a4c9-45d4-a16f-406866b7fc05", "namespacedname": "agent/my-agent", "kind": "StatefulSet", "err": "no new version (current: \"v12.2.3\", next: \"v12.2.3\")"}
+```
+
+</TabItem>
+<TabItem label="systemd " options="Teleport Cloud">
+
+If you changed the agent user to run as non-root, create
+`/etc/teleport-upgrade.d/schedule` and grant ownership to your Teleport user.
+Else, you can skip this step:
+
+```code
+$ mkdir -p /etc/teleport-upgrade.d/
+$ touch /etc/teleport-upgrade.d/schedule
+$ chown <your-teleport-user> /etc/teleport-upgrade.d/schedule
+```
+
+Install the `teleport-ent-updater` package (note: your teleport agent will be restarted during install):
+
+```code
+$ apt install teleport-ent-updater
```

Comment:
@marcoandredinis @hugoShaka Do we need to update our Discover install script and auto-discovery install script to use this package name when auto-upgrades env. var is set? IIRC we only change the repository channel right now.

Reply:
Yup, I thought this was part of [this ticket](https://github.com/gravitational/teleport/issues/22731) but it's not.

Reply:
Should we include `teleport-updater` as a dependency of `teleport` apt/yum package?
(and `teleport-ent-updater` a dep of `teleport-ent`)

If we do so, then we would only need to tell users to configure and then `systemctl enable --now` the service.
I do understand that this is one more binary we add to the base installation but it seems we are going to allow all users to run the upgrader, so we should be ok with this being on user's machine.

I'll open the PR to install the package on our install scripts but please let me know what you think.

Reply:
I think we do it the other way: if you install `teleport-ent-updater`, you also get `teleport-ent` with the right systems unit and started. cc @fspmarshall for confirmation

We don't have a `teleport-updater` OSS package yet 😢 .  Everything is OSS-friendly but the packaging didn't happen because we were short on time.